src/main/java/com/mpsp/cc_auth_service/service/impl/AuthServiceImpl.java

```
@@ -143,15 +152,34 @@ public LoginResponse refreshToken(final String refreshToken) {
   }
 
   @Override
-  @Transactional(readOnly = true)
+  @Transactional(readOnly = false)
   public void sendResetPasswordEmail(String email) {
-    userService.findByEmail(email);
+    User user = userService.findByEmail(email);
+
+    if (user == null) {
+      throw new GlobalExceptionHandler.GenericException("User not found");
+    }
 
-    awsService.sendEmail(senderEmail, email, "cc_reset_password", Map.of("link", resetPasswordUrl));
+    Optional<ResetPassword> existingTokenOpt = resetPasswordRepo.findByUserIdAndLinkSent(user.getUserId());
+    if (existingTokenOpt.isPresent()) {
+      ResetPassword existingToken = existingTokenOpt.get();
+      if (Duration.between(existingToken.getCreatedAt(), LocalDateTime.now()).toHours() < 1) {
+        throw new GlobalExceptionHandler.GenericException("A password reset link has already been sent. Please try again later.");
+      }
```

Same. This is a specific problem, can we throw a specific exception.